app/assets/src/components/ui/controls/SampleTypeSearchBox.jsx

```diff
@@ -0,0 +1,93 @@
+import React from "react";
+import { groupBy } from "lodash/fp";
+
+import PropTypes from "~/components/utils/propTypes";
+
+import LiveSearchPopBox from "./LiveSearchPopBox";
+
+class SampleTypeSearchBox extends React.Component {
+  handleSearchTriggered = query => {
+    const matchSampleTypes = sampleType => {
+      // Match chars in any position. Good for acronyms.
+      const regex = new RegExp(query.split("").join(".*"), "gi");
+      if (regex.test(sampleType.name)) {
+        return true;
+      }
+      return false;
+    };
+    const matchedSampleTypes = this.props.sampleTypes.filter(matchSampleTypes);
+
+    const isSuggested = sampleType => {
+      // Insects only get their own types
+      if (this.props.isInsect) {
+        return sampleType.insect_only ? "suggested" : "donotshow";
```

Comment:
These string should probably be constants.

Reply:
No problem. Side note: it's funny how we are more strict about constants in javascript than in python.

Reply:
I am guessing you meant ruby... when i notice the pattern i try to push for it on both. Not sure if I have been consistent though :P

Reply:
No, I meant python, in the pipeline. It's full of magic strings. I think in ruby we consistently use constants.